tests/dtm/test_rgw_s3_process_restart.py

```diff
@@ -0,0 +1,287 @@
+            self.log.info("Test Failure observed, collecting support bundle")
+            path = os.path.join(LOG_DIR, LATEST_LOG_FOLDER)
+            resp = support_bundle_utils.collect_support_bundle_k8s(
+                local_dir_path=path, scripts_path=const.K8S_SCRIPTS_PATH)
+            assert_utils.assert_true(resp)
+        if self.iam_user:
+            self.log.info("Cleanup: Cleaning created IAM users and buckets.")
+            resp = self.ha_obj.delete_s3_acc_buckets_objects(self.iam_user)
+            assert_utils.assert_true(resp[0], resp[1])
+        if os.path.exists(self.test_dir_path):
+            system_utils.remove_dirs(self.test_dir_path)
+
+    @pytest.mark.lc
+    @pytest.mark.dtm
+    @pytest.mark.tags("TEST-42244")
+    def test_read_during_rgw_svc_restart(self):
+        """Verify READs during rgw_s3 service restart using pkill."""
+        self.log.info("STARTED: Verify READs during rgw_s3 service restart using pkill")
+        log_file_prefix = 'test-42244'
+        que = multiprocessing.Queue()
+
+        self.log.info("Step 1: Perform write operation for background read")
+        self.dtm_obj.perform_write_op(bucket_prefix=self.bucket_name,
+                                      object_prefix=self.object_name,
+                                      no_of_clients=self.test_cfg['clients'],
+                                      no_of_samples=self.test_cfg['samples'],
+                                      obj_size=self.test_cfg['size'],
+                                      log_file_prefix=log_file_prefix, queue=que)
+        resp = que.get()
+        assert_utils.assert_true(resp[0], resp[1])
+        workload_info = resp[1]
+        self.log.info("Step 2: Perform Read Operations on the data written in step 1 in background")
+        proc_read_op = multiprocessing.Process(target=self.dtm_obj.perform_ops,
+                                               args=(workload_info, que, False, True, True))
+        proc_read_op.start()
+
+        self.log.info("Step 3: Perform rgw_s3 Process Restart for %s times During Read "
+                      "Operations", DTM_CFG["rgw_restart_cnt"])
+        resp = self.dtm_obj.process_restart(master_node=self.master_node_list[0],
+                                            health_obj=self.health_obj,
+                                            pod_prefix=const.SERVER_POD_NAME_PREFIX,
+                                            container_prefix=const.RGW_CONTAINER_NAME,
+                                            process=self.rgw_process, check_proc_state=False,
+                                            restart_cnt=DTM_CFG["rgw_restart_cnt"])
+        assert_utils.assert_true(resp, "Failure observed during process restart/recovery")
+
+        self.log.info("Step 4: Wait for READ operation to complete.")
+        if proc_read_op.is_alive():
+            proc_read_op.join()
+        resp = que.get()
+        assert_utils.assert_true(resp[0], resp[1])
```

Comment:
aren't we expecting it to fail and then retried from client? So that will be taken care by retry count?

Reply:
It will be retried internally as we are providing max_retries option to s3bench. So we expect all the IOs to be passed.

Reply:
It will be retried internally as we are providing max_retries option to s3bench. So we expect all the IOs to be passed.